option.go

```diff
@@ -99,6 +102,27 @@ func BlockHook(blockHook BlockHookFunc) Option {
 	}
 }
 
+// SegmentBlockHook adds a hook that runs when a block is received along with a SegmentSyncActions.
+// Note that if segmented sync is disabled, calls on SegmentSyncActions will have no effect.
+// See: SegmentSyncActions, SegmentDepthLimit.
+func SegmentBlockHook(sbh SegmentBlockHookFunc) Option {
```

Comment:
Would it be better to combine `SegmentBlockHook` and `SegemntDepthLimit` into one option since these need to be specified together?
```go
func SegmentBlockHook(sbh SegmentBlockHookFunc, depth int64) Option {
    ...
}
```

Reply:
I wondered about doing this and settled on separate options. The rationale is:
* having separate options means they can be individually overridden, whereas if they are coupled the caller has to know both in order to override one.
* in the future if we come up with some default for the values that has segmentation enabled by default the options API would look more reasonable as separate options.
* it makes sense to separate them in the scoped variation, since either could be populated from a default set in `Subscriber` level, which then made the API look nicer (to me) if they are also split here.

Another option is to have both APIs. Happy to change it if this is a blocker.